### ISSUE TITLE
[FavoriteMedia]: CAT ID is give even if it's already used

### ISSUE THREAD

**Issue by AlphaGaming7780** [Dastan21/BDAddons#223]
### Discord Build

Stable

### Discord Version

Stable 235965 (40a35e6)

### Plugin Version

1.8.8

### Steps To Reproduce

Just create a lot of categories, and also categories in categories, move them (before / after) or look the screens.

### Describe the bug

The plugins only read the last entry in the config file for the category ID when we create a new one, or something's like that.
That make the plugins create new categories with the same ID as others.

### Screenshots

Here it's the same ID
![image](https://github.com/Dastan21/BDAddons/assets/69048433/a0d489d6-414f-4bff-bae6-e14d412c5c07)

Here the new CAT get a new ID
![image](https://github.com/Dastan21/BDAddons/assets/69048433/0d7ae772-e5fb-4aba-902f-99c565073aea)

And here it's a big mess
![image](https://github.com/Dastan21/BDAddons/assets/69048433/e3310efe-029e-48f4-958b-c720fd107c23)

**Comment by AlphaGaming7780** [comment]
So I try to fix this issues and here is what I have done.

I add/edit : 
```js
  /**
   * Try to find an ID for new CAT
   * @param {Array} categories the list of all the categories
   * @returns {Int}
   */
  function categoryGetNewID (categories) {
    // This for try to find an unused ID. Maybe remove that because it's a for and not optimize
    for(id = 1; id <= categories.length; id++) { // Start at one, cause you made +1 before, so it's not possible to create a CAT with the ID 0
      if(categories.find(c => c.id === id) === undefined) {
        return id
      }
    }
    return categories.length + 1 // Add one cause you add 1 before
  }

  function createCategory (type, { name, color }, categoryId) {
    const res = categoryValidator(type, name, color)
    if (res.error) {
      console.error('[FavoriteMedia]', res.error)
      Toasts.error(res.message)
      return false
    }

    res.categories.push({ id: categoryGetNewID(res.categories), name, color, category_id: categoryId }) // Previous line : res.categories.push({ id: ((res.categories.slice(-1)[0] && res.categories.slice(-1)[0].id) || 0) + 1, name, color, category_id: categoryId })
    Utilities.saveData(config.name, type, res)

    Toasts.success(labels.category.success.create)
    return true
  }
```

Result :
I discover that the CAT ID 12 is free, and it successfully creates a CAT with the ID 26 instead of 18 or something's else.
![image](https://github.com/Dastan21/BDAddons/assets/69048433/a3303f11-e454-4599-ac48-e9ce6baa6ccb)

**Comment by Dastan21** [comment]
Thanks for your help. Fixed in 1.8.9
